pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -355,13 +355,15 @@ func (p *ldapProvider) searchUser(name string, config *v3.LdapConfig, lConn *lda
 	for _, attr := range srchAttributes {
 		srchAttrs += fmt.Sprintf("(%v=%v*)", attr, name)
 	}
-	query += srchAttrs + "))"
+	// The user search filter will be added as another and clause
+	// and is expected to follow ldap syntax and enclosed in parenthesis
+	query += srchAttrs + ")" + config.UserSearchFilter + ")"
 	logrus.Debugf("%s searchUser query: %s", p.providerName, query)
 	return p.searchLdap(query, p.userScope, config, lConn)
 }
 
 func (p *ldapProvider) searchGroup(name string, config *v3.LdapConfig, lConn *ldapv2.Conn) ([]v3.Principal, error) {
-	query := fmt.Sprintf("(&(%v=*%v*)(%v=%v))", config.GroupSearchAttribute, name, ObjectClass, config.GroupObjectClass)
+	query := "(&(" + ObjectClass + "=" + config.GroupObjectClass + ")(" + config.GroupSearchAttribute + "=*" + name + "*)" + config.GroupSearchFilter + ")"
```

Comment:
Curious, do we have any kind of safe query verify functions that can be used here? Since we are using the string input from the user directly here, it can potentially be used as an attack vector.

Reply:
I am not aware of a safe query verify function. This can be only configured by an admin while setting up auth. And that query is done against the admin's ldap server. It only adds an additional filter to the output being returned. So I don't see how it can be used as an attack against rancher either.

Reply:
You are right, this is an admin config. Probably, it my comments are not relevant in this context. Anytime I see any text input from user being used, I get paranoid.